### ISSUE TITLE
[java] MissingSerialVersionUID rule does not seem to catch inherited classes

### ISSUE THREAD

**Issue by lgoldstein** [pmd/pmd#1078]
Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core].

**Affects PMD Version:** 

6.2.0

**Rule:**

MissingSerialVersionUID

**Description:**

**Code Sample demonstrating the issue:**

```java
public class Foo implements Serializable {
     private static final long serialVersionUID = 1234567L;
}

// PMD does not complain about this class
public class Bar extends Foo {
    // ... no serialVersionUID value defined ...
}
```

see [old rejected patch for same issue](https://sourceforge.net/p/pmd/patches/249/)

**Running PMD through:** *[Maven]*
